api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -117,6 +117,9 @@ public static ByteBuffer toByteBuffer(Type.TypeID typeId, Object value) {
         return (ByteBuffer) value;
       case DECIMAL:
         return ByteBuffer.wrap(((BigDecimal) value).unscaledValue().toByteArray());
+      case UNKNOWN:
+        // underlying type not known
```

Comment:
the lower/upper bound is converted in https://github.com/apache/iceberg/blob/5ab73bfdca075533a9e15e04e6fdbc5ef1f72e9d/core/src/main/java/org/apache/iceberg/util/ManifestFileUtil.java#L51-L52 but we can't convert it to its underlying type, since the type isn't known